src/Module.Server/Rewards/CrpgAgent.cs

```diff
@@ -0,0 +1,22 @@
+﻿namespace Crpg.Module.Rewards;
+internal class CrpgAgent
```

Comment:
The name is very misleading since it's completely different from a native agent. Maybe you could name it `AgentHitRegistry` or something like that.